### PR TITLE
specfem3d-globe: Add new package 

### PR DESCRIPTION
**Pull request by t-karatsu**
Add new package `specfem3d-globe`.
Ref: https://github.com/geodynamics/specfem3d_globe

Currently, we have confirmed that only the Fujitsu compiler can be built. Since the purpose is to be able to see this recipe in the latest Spack, I submitted a PR first, although it contains unresolved following issues.
We will update as soon as the problem is resolved.

 - about issue
When building with the gcc compiler,'Werror' is added to FFLAGS.
In the case of using the gcc compiler and the default simulation settings, 
there is the process which always causes array out-of-bounds reference error,
and this is processed as error instead of warning.
So, temporarily set the gcc compiler as a conflict.
This issue is queried in the following:
https://github.com/geodynamics/specfem3d_globe/issues/717

### CODE REVIEW

**Comment by t-karatsu** [comment]
>When building with the gcc compiler,'Werror' is added to FFLAGS.
In the case of using the gcc compiler and the default simulation settings,
there is the process which always causes array out-of-bounds reference error,
and this is processed as error instead of warning.

The developer of `specfem3d-globe` has contacted me that in the current version we can remove `-Werror` to avoid the problem. So, I fixed this PR. Version 8.0.0 seems to fix this issue, so this patch will be applied up to 7.0.2.

File: var/spack/repos/builtin/packages/specfem3d-globe/package.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Specfem3dGlobe(AutotoolsPackage):
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/specfem3d-globe/package.py]: Can we inherit from `CudaPackage` and remove the dependency on `cuda` and the `cuda` variant below?

File: var/spack/repos/builtin/packages/specfem3d-globe/package.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            description="Treat REAL as double precision")
+
+    depends_on('mpi')
+    depends_on('cuda', when='+cuda')
+    depends_on('opencl', when='+opencl')
+
+    # When building with the gcc compiler,'Werror' is added to FFLAGS.
+    # In the case of using the gcc compiler and the default simulation
+    # settings, there is the process which always causes
+    # array out-of-bounds reference error.
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/specfem3d-globe/package.py:32]: :scream:

**Review by alalazo — CHANGES_REQUESTED** [review]
The recipe LGTM. I have one question on using `CudaPackage` and a more general question of whether something where:

> there is the process which always causes array out-of-bounds reference error

should be shipped at all with Spack. Should we wait for 8.0.0 to be released? What is the timeframe?

File: var/spack/repos/builtin/packages/specfem3d-globe/package.py (hunk 1 above)
Comment (t-karatsu) [reply var/spack/repos/builtin/packages/specfem3d-globe/package.py]: I fixed it! There is no environment to check the Cuda function, so please let me know if you have any problems.

File: var/spack/repos/builtin/packages/specfem3d-globe/package.py (hunk 2 above)
Comment (t-karatsu) [reply var/spack/repos/builtin/packages/specfem3d-globe/package.py:32]: 😱 Version 8.0.0 seems to solve this problem, so can 7.0.2 be allowed to remove `-Werror`?
The developer has contacted me that this countermeasure is possible:
https://github.com/geodynamics/specfem3d_globe/issues/717

**Comment by t-karatsu** [comment]
@alalazo 
> should be shipped at all with Spack. Should we wait for 8.0.0 to be released? What is the timeframe?

Well... some members want to use this recipe, so I hope it will be merged as soon as possible.
Could you please let me know if I have any necessary corrections or comments?

**Review by adamjstewart — APPROVED** [review]

**Review by alalazo — APPROVED** [review]

**Comment by alalazo** [comment]
> Well... some members want to use this recipe, so I hope it will be merged as soon as possible.
Could you please let me know if I have any necessary corrections or comments?

@t-karatsu Done. To be clear, the recipe looks good. I was just worried that the compiler reported out of bounds error because I'm pretty sure if they are reported they are there :slightly_smiling_face:

**Pull request merged**